[KOGITO-2550] - Fixed vet test

Kaitou786 opened the pull request:
See: https://issues.redhat.com/browse/KOGITO-2550
Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

Kaitou786 commented on hack/go-vet.sh:
```diff
@@ -14,19 +14,16 @@
 
 
-if [[ -z ${CI} ]]; then
-    ./hack/go-mod.sh
```
Because of this condition the script was not executing. In circle CI this is an enviornment variable set to `true`

ricardozanini replied:
You want this to be executed to catch when someone send a PR without running `make mod`?

Kaitou786 replied:
Not just this step, but all the steps were in this condition like operator-sdk generate k8s and crds, which were not executed on this CircleCI

Kaitou786 replied:
Because of this condition the script `go-vet.sh` was only running a single command `go vet ./...`

Kaitou786 commented:
Because the steps were missed(due to that if condition) there are some changes in the files. I changed those in https://github.com/kiegroup/kogito-cloud-operator/pull/412. After that is merged I can rebase and this one will succeed

ricardozanini commented:
@Kaitou786 this one depends on #412?

Kaitou786 commented:
@ricardozanini on the master branch itself if you run `make test` you'll see the changes on the CRDs, I first noticed this when sending #412 . So after the #412 is merged the lint stage here will pass (After Rebase)

spolti commented:
can you verify why lint in failing?

Kaitou786 commented:
@spolti yes, the lint is failing because the change is detected in the crds. Previously this check was not able to detect the changes as the operator-sdk command was not executing,  because of that some commit got merged in the master which should have also changed the crds. I changed added those crds in #412 After the rebase the lint over here will succeed.

Kaitou786 commented:
@ricardozanini I made some changes can you review again?

ricardozanini commented on .circleci/config.yml:
```diff
@@ -9,27 +9,37 @@ jobs:
+          command: |
+            ./hack/go-vet.sh
+            changed_files=$(git status -s | grep -v 'go.mod\|go.sum' || :)
+            openapi_changed_files=$(git status -s | grep zz_generated || :)
```
Just a nitpick in here. But `zz_generated` is also a prefix for the controller-gen object deepcopy generation. So the variable name could be `generated_changed_files`. :D

Kaitou786 replied:
What do you suggest should we handle only the openapi files or for both generated files?

ricardozanini replied:
I think both. DeepCopy also changes if have a `*_types` change. Your script already will caught that, just need to change the var name.

Kaitou786 replied:
Updated
